source/val/validate_memory.cpp

```diff
@@ -323,6 +324,34 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
              << "Classes: Workgroup, CrossWorkgroup, Private, and Function";
     }
   }
+
+  // Vulkan spec 14.5.2
+  if (spvIsVulkanEnv(_.context()->target_env) &&
+      storage_class == SpvStorageClassUniformConstant) {
+    auto variable_type = _.FindDef(result_type->GetOperandAs<uint32_t>(2));
+    auto variable_type_opcode = variable_type->opcode();
+
+    // If the variable is actually an array extract the element type.
+    if (variable_type_opcode == SpvOpTypeArray ||
+        variable_type_opcode == SpvOpTypeRuntimeArray) {
```

Comment:
According the SPIR-V environment for Vulkan, only SSBOs (StorageBuffer with Block or Uniform with BufferBlock) can have a runtime array so drop this condition.

Reply:
Done